docs/pages/web/textfield.js

```diff
@@ -700,6 +698,78 @@ function TextFieldExample() {
     />
   );
 }
+`}
+          />
+        </MainSection.Subsection>
+        <MainSection.Subsection
+          title="Maximum length"
+          columns={2}
+          description={`Textfield supports the native [maxlength](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/maxlength) input attribute. \`maxLength\` sets the maximum number of characters allowed to be entered by the user in Textfield.
+          
+For controlled TextFields, if the initial value in the external state exceeds the \`maxLength\` value, Textfield will display the long value; however, the user won't be able to add additional characters and they will only be allowed to delete characters. 
+          
+When \`maxLength\` is passed to TextField, the component displays a character counter as well as an [error Status](/status) when the user reaches or the controlled value exceeds the maximum length of characters. 
+
+The first example shows an empty Textfield with \`maxLength\` set to 20 characters. The second example shows the error Status. The example below should be prevented as it shows Textfield displaying the error Status when the value lenght is exceeded.`}
```

Comment:
Typo: length

Not clear what "The example below should be prevented as it shows Textfield displaying the error Status when the value lenght is exceeded." is hoping to convey. 

Is it saying that you should prevent the user from being able to type after the limit is reached?

Reply:
We shouldn't set initial values from the parent component state that already exceed the maxLength. I'm not sure if this could be possible if we pre-populate fields...

Reply:
Ahh ok, maybe then 

"Whenever possible, avoid setting initial values from the parent component's state that already exceed the maxLength"?

Reply:
Would we not want to show the error state if the initial value exceeds the max length?

Reply:
This is on me and in retrospect, I think we should display a warning icon as opposed to an error icon. Based on the current functionality, we don't allow the input to go beyond the character limit.

Reply:
Input from the user, yes. But it's possible for the developer to specify an initial state for the input that exceeds the limit, which we would want to communicate. (This initial value may be determined dynamically, so just throwing an actual error wouldn't be a good idea.)